lib/route/link/macvlan.c

```diff
@@ -155,9 +227,23 @@ static int macvlan_put_attrs(struct nl_msg *msg, struct rtnl_link *link)
 	if (mvi->mvi_mask & MACVLAN_HAS_FLAGS)
 		NLA_PUT_U16(msg, IFLA_MACVLAN_FLAGS, mvi->mvi_flags);
 
-	nla_nest_end(msg, data);
+	if (mvi->mvi_mask & MACVLAN_HAS_MACDATA) {
+		NLA_PUT_U32(msg, IFLA_MACVLAN_MACADDR_MODE, mvi->mvi_macmode);
+		datamac = nla_nest_start(msg, IFLA_MACVLAN_MACADDR_DATA);
+		if (!datamac)
+			goto nla_put_failure;
+
+		for (i = 0; i < mvi->mvi_maccount; i++) {
+			NLA_PUT_ADDR(msg, IFLA_MACVLAN_MACADDR,
+				     mvi->mvi_macaddr[i]);
+		}
+	}
 
 nla_put_failure:
+	if (datamac)
```

Comment:
seems macvlan_put_attrs() should return -NLE_NOMEM in case of a nla_put_failure and not 0. Can you fix that too?